src/Frontend/Modules/Mailmotor/EventListener/SubscribeFromFormBuilderSubmittedForm.php

```diff
@@ -22,60 +23,58 @@
-        ModulesSettings $modulesSettings
-    ) {
+    public function __construct(Subscriber $subscriber, ModulesSettings $modulesSettings)
+    {
         $this->subscriber = $subscriber;
         $this->modulesSettings = $modulesSettings;
     }
 
-    public function onFormBuilderSubmittedEvent(
-        FormBuilderSubmittedEvent $event
-    ): void {
-        if ($this->modulesSettings->get('Mailmotor', 'automatically_subscribe_from_form_builder_submitted_form', false)) {
-            $form = $event->getForm();
-            $data = $event->getData();
-            $email = null;
+    public function onFormBuilderSubmittedEvent(FormBuilderSubmittedEvent $event): void
+    {
+        if (!$this->modulesSettings->get(
+            'Mailmotor',
+            'automatically_subscribe_from_form_builder_submitted_form',
+            false
+        )) {
+            return;
+        }
+        $form = $event->getForm();
+        $data = $event->getData();
+        $email = null;
 
-            // Check if we have a replyTo email set
-            foreach ($form['fields'] as $field) {
-                if (array_key_exists('reply_to', $field['settings']) &&
-                    $field['settings']['reply_to'] === true
-                ) {
-                    $email = unserialize($data[$field['id']]['value']);
-                }
+        // Check if we have a replyTo email set
+        foreach ($form['fields'] as $field) {
+            if (array_key_exists('reply_to', $field['settings']) &&
+                $field['settings']['reply_to'] === true
+            ) {
+                $email = unserialize($data[$field['id']]['value'], ['allowed_classes' => false]);
             }
+        }
 
-            // Define language
-            $language = array_key_exists('language', $form)
-                ? $form['language'] : $this->modulesSettings->get('Core', 'default_language', 'en');
-
-            // We subscribe the replyTo email
-            try {
-                // Does email exists or not in our mailing list
-                $exists = (bool) $this->subscriber->exists($email);
+        $language = $form['language'] ?? $this->modulesSettings->get('Core', 'default_language', 'en');
 
-                // We only need to subscribe when not exists
-                if (!$exists) {
-                    $this->subscriber->subscribe(
-                        $email,
-                        $language,
-                        [],
-                        [],
-                        false // will ignore double-optin and so subscribes the user immediately
-                    );
-                }
-            } catch (NotImplementedException $e) {
-                // We do nothing as fallback when no mail-engine is chosen in the Backend
+        // We subscribe the replyTo email
+        try {
+            if (!$this->subscriber->exists($email)) {
+                // @TODO check if we are allowed to do this like this according to GDPR
```

Comment:
You can't do that. The only way you could implement this, if you would add a checkbox in the form that explains to the user on which list he/she will be subscribed, the interval, ... 
So doing this automatically should not be in the default implementation of Fork, so I advice to remove all related code.

Reply:
solved in a7352acf1